train.py

```diff
@@ -370,82 +329,39 @@ def eval(
         for tracker_name in ["training", "validation", "testing"]
     )
 
-    epoch_progress = Progress(
-        "{task.description}",
-        SpinnerColumn(),
-        BarColumn(),
-        TextColumn("{task.percentage:>3.0f}%"),
-        metric_tracker_train.get_metric_text_column(),
-    )
-
-    train_progress_dict = {
-        **dict(description="[green]Training", total=len(train_set_loader)),
-        **metric_tracker_train.get_current_iteration_metric_text_column_fields(),
-    }
-
-    val_progress_dict = {
-        **dict(description="[yellow]Validation", total=len(val_set_loader)),
-        **metric_tracker_val.get_current_iteration_metric_text_column_fields(),
-    }
-
-    train_epoch_progress = epoch_progress.add_task(**train_progress_dict)
-    val_epoch_progress = epoch_progress.add_task(**val_progress_dict)
-
-    for task in epoch_progress.tasks:
-        print(task)
-
-    overall_progress = Progress()
-
-    total_iters = (args.max_epochs - 1 - start_epoch) * (
-        len(train_set_loader) + len(val_set_loader)
-    ) + len(test_set_loader) * args.test
-
-    overall_task = overall_progress.add_task(
-        "Experiment Progress",
-        completed=(start_epoch + 1) * (len(train_set_loader) + len(val_set_loader)),
-        total=total_iters,
-    )
-
-    progress_table = Table.grid()
-    progress_table.add_row(
-        Panel.fit(
-            overall_progress,
-            title="Experiment Progress",
-            border_style="green",
-            padding=(2, 2),
+    progress_reporter = PrettyProgressReporter(
+        metric_trackers=(metric_tracker_train, metric_tracker_val, metric_tracker_test),
+        set_size_list=(
+            len(train_set_loader),
+            len(val_set_loader),
+            len(test_set_loader),
         ),
+        max_epochs=args.max_epochs,
+        start_epoch=start_epoch,
+        test=args.test,
     )
-    progress_table.add_row(
-        Panel.fit(epoch_progress, title="[b]Jobs", border_style="red", padding=(1, 2))
-    )
-    progress_table.add_row(metric_tracker_train.per_epoch_table)
-    progress_table.add_row(metric_tracker_val.per_epoch_table)
 
     train_iterations = 0
 
-    with Live(progress_table, refresh_per_second=10) as interface_panel:
+    with Live(
```

Comment:
This is good for now but it would be better in future if:
- we could provide a more generic `with` handler (not specific to rich) 
- we could wrap rich in a builder 
- the user could then choose the type of metric tracker (e.g. plain printing, rich, tqdm) from the train script

Reply:
I agree. This would be really cool.

Reply:
I'll open an issue for this then! :)